internal/data-importer/cmd/psmdb/import.go

```diff
@@ -0,0 +1,207 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      psmdbRestoreName,
+			Namespace: namespace,
+		},
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, c, secret, func() error {
+		secret.Type = corev1.SecretTypeOpaque
+		secret.StringData = map[string]string{
+			"AWS_ACCESS_KEY_ID":     accessKeyId,
+			"AWS_SECRET_ACCESS_KEY": secretAccessKey,
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func runPSMDBRestoreAndWait(
+	ctx context.Context,
+	c client.Client,
+	namespace string,
+	dbName string,
+	psmdbRestoreName string,
+	backupPath string,
+	bucket, endpoint, region string,
+) error {
+	// parse the backup path to extract prefix and destination.
+	backupPath = strings.Trim(backupPath, "/")
+	split := strings.Split(backupPath, "/")
+	prefix := strings.Join(split[:len(split)-1], "/")
+	destination := fmt.Sprintf("s3://%s/%s", bucket, backupPath)
+
+	psmdbRestore := &psmdbv1.PerconaServerMongoDBRestore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      psmdbRestoreName,
+			Namespace: namespace,
+		},
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, c, psmdbRestore, func() error {
+		psmdbRestore.Spec = psmdbv1.PerconaServerMongoDBRestoreSpec{
+			ClusterName: dbName,
+			BackupSource: &psmdbv1.PerconaServerMongoDBBackupStatus{
+				Type:        pbmdefs.LogicalBackup,
+				Destination: destination,
+				S3: &psmdbv1.BackupStorageS3Spec{
+					Bucket:            bucket,
+					Region:            region,
+					EndpointURL:       endpoint,
+					CredentialsSecret: psmdbRestoreName,
+					Prefix:            prefix,
+				},
+			},
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to create or update PSMDB restore: %w", err)
+	}
+
+	// wait for it to be completed.
+	return wait.PollUntilContextCancel(ctx, 5*time.Second, true, func(ctx context.Context) (bool, error) {
+		psmdbRestore := &psmdbv1.PerconaServerMongoDBRestore{}
+		if err := c.Get(ctx, types.NamespacedName{
+			Name:      psmdbRestoreName,
+			Namespace: namespace,
+		}, psmdbRestore); err != nil {
+			return false, fmt.Errorf("failed to get PSMDB restore %s: %w", psmdbRestoreName, err)
+		}
+		return psmdbRestore.Status.State == psmdbv1.RestoreStateReady, nil
+	})
+}
+
+func cleanup(
+	ctx context.Context,
+	c client.Client,
+	namespace string,
+	psmdbRestoreName string,
+) error {
+	// delete S3 secret.
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      psmdbRestoreName,
+			Namespace: namespace,
+		},
+	}
+	if err := c.Delete(ctx, secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to delete S3 credentials secret %s: %w", psmdbRestoreName, err)
+	}
+
+	// delete PSMDB restore.
+	restore := &psmdbv1.PerconaServerMongoDBRestore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      psmdbRestoreName,
+			Namespace: namespace,
+		},
+	}
+	if err := c.Delete(ctx, restore); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to delete PSMDB restore %s: %w", psmdbRestoreName, err)
+	}
```

Comment:
as agreed in the pxc PR, let's keep the restore object for visibility

Reply:
done